Database/Stored Procedures/add_or_update_festival.sql

```diff
@@ -0,0 +1,73 @@
+/*==============================================================*/
+/* DBMS name:		FestiBase									*/
+/* PDM version:		6											*/
+/* Last edited:		20-12-2017									*/
+/* Edited by:		Mariusz Blautzik							*/
+/* Procedure:		Insert + Update FESTIVAL					*/
+/*==============================================================*/
+
+DROP PROCEDURE IF EXISTS sp_add_or_update_festival
+GO
+CREATE PROCEDURE sp_add_or_update_festival
+   @insert				 BIT,
+   @festival_number      INT = NULL,
+   @organisation_number  INT,
+   @name                 VARCHAR(50),
+   @start_date           DATETIME,
+   @end_date             DATETIME,
+   @location             VARCHAR(50),
+   @token_price          MONEY
+   
+
+AS
+BEGIN
+	BEGIN TRY
+
+		IF (@insert = 1)
+		BEGIN
+			INSERT INTO FESTIVAL(organisation_number, "name", "start_date", end_date, "location", token_price)
+			VALUES (@organisation_number, 
+					@name, 
+					@start_date, 
+					@end_date, 
+					@location, 
+					@token_price)
+
+		END
+	ELSE 
+			BEGIN
+				IF (@festival_number = null)
+					BEGIN
+						;THROW 50000, '@festival_number cannot be NULL if an UPDATE is to be commenced', 1
+					END
+
```

Comment:
voeg deze error toe
ELSE IF NOT EXISTS (SELECT * FROM festival WHERE festival_number = @festival_number)
BEGIN
;THROW 50000, 'This festival does not exist', 1
END